### PR TITLE
feat(insights): hide "other" breakdown

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Not everyone is happy with the "Other" column.

## Changes

Adds a way to remove the column when you edit the chart.

I had to make NULL / None handling for breakdowns very explicit as well. Because of our column materialization, we must also treat `""` as None.

## How did you test this code?

Backend tests

### CODE REVIEW

**Comment by webjunkie** [comment]
Reminds me of Kibana:

image

Maybe we'll also do a toggle?

File: posthog/queries/trends/sql.py [hunk 1]
```diff
@@ -98,7 +98,6 @@
 """
 
 TOP_ELEMENTS_ARRAY_OF_KEY_SQL = """
-SELECT groupArray(value) FROM (
```
Comment (mariusandra) [inline posthog/queries/trends/sql.py:101]: Running through `groupArray()` removes the rows where this field is `None`. We however do want to know if `None` is part of the returned fields or not.

File: posthog/queries/breakdown_props.py [hunk 2]
```diff
@@ -233,7 +233,12 @@ def get_breakdown_prop_values(
         query_type="get_breakdown_prop_values",
         filter=filter,
         team_id=team.pk,
-    )[0][0]
+    )
+
+    if filter.using_histogram:
+        return response[0][0]
+    else:
+        return [row[0] for row in response]
```
Comment (mariusandra) [inline posthog/queries/breakdown_props.py:241]: There are two queries here: 1) integer histogram breakdown, 2) string top results breakdown.

The response of the second query changed in order to also return nulls. More on it below.

File: posthog/queries/trends/breakdown.py [hunk 3]
```diff
@@ -83,6 +84,8 @@
 
 BREAKDOWN_OTHER_STRING_LABEL = "$$_posthog_breakdown_other_$$"
 BREAKDOWN_OTHER_NUMERIC_LABEL = 9007199254740991  # pow(2, 53) - 1, for JS compatibility
+BREAKDOWN_NULL_STRING_LABEL = "$$_posthog_breakdown_null_$$"
+BREAKDOWN_NULL_NUMERIC_LABEL = 9007199254740990  # pow(2, 53) - 2, for JS compatibility
```
Comment (mariusandra) [inline posthog/queries/trends/breakdown.py:88]: I'm not a giant fan of magic strings, but for now I consider this to be good enough... Or at least "helps us today and doesn't block us from adding a proper fix in the future".

**Comment by mariusandra** [comment]
Finally ready for a review

File: frontend/src/scenes/trends/Trends.tsx [hunk 4]
```diff
@@ -66,27 +66,43 @@ export function TrendInsight({ view, context }: Props): JSX.Element {
...
+                                    bordered
+                                    checked={!breakdown.breakdown_hide_other_aggregation}
+                                    onChange={() =>
+                                        updateBreakdown({
+                                            ...breakdown,
+                                            breakdown_hide_other_aggregation:
+                                                !breakdown.breakdown_hide_other_aggregation,
+                                        })
+                                    }
+                                    label='Group renaming values under "Other"'
```
Comment (webjunkie) [inline frontend/src/scenes/trends/Trends.tsx:102]:
```suggestion
                                    label='Group remaining values under "Other"'
```

**Review by webjunkie — APPROVED** [review]
Ah, I found it, but only with HogQL trends turned off. Is that as intended?

Otherwise, seems to work alright.

**Comment by mariusandra** [comment]
Thanks for the review!

Yes, for now this is only implemented for the non-HogQL trends. I have added "Breakdown controls" as one of the things to fix for the [rewrite](https://github.com/PostHog/meta/issues/130).

I also put an end to the frontend innovation, and moved the button into the sidebar with the other editor filters:



There will likely be some changed snapshots because of this.

One of the next things we can add here is the option to toggle exactly how many breakdown values are displayed, and which ones, but that's for future PRs.

**Comment by thmsobrmlr** [comment]
Fly-by suggestion: We already have some options relating to breakdowns in the `...`-menu. Should we put the toggle there? I'd also add the breakdown limit into it.

Screenshot 2023-12-19 at 09 43 16

**Comment by mariusandra** [comment]
Something like this?

**Comment by thmsobrmlr** [comment]
> Something like this?  

Exactly this.

File: frontend/src/scenes/insights/filters/BreakdownFilter/BreakdownTagMenu.tsx [hunk 5]
```diff
@@ -85,6 +91,58 @@ export const BreakdownTagMenu = (): JSX.Element => {
...
+                            status="stealth"
+                            active={histogramBinsUsed}
+                            fullWidth
+                        >
+                            Breakdown limit:{' '}
+                            <LemonInput
+                                min={1}
+                                value={breakdownLimit}
+                                onChange={(newValue) => {
+                                    setBreakdownLimit(newValue ?? 25)
```
Comment (webjunkie) [inline frontend/src/scenes/insights/filters/BreakdownFilter/BreakdownTagMenu.tsx:137]: What doesn't work here for me is typing a value and then focusing somewhere else, thereby closing the menu. That doesn't seem to trigger onChange.

Comment (mariusandra) [reply frontend/src/scenes/insights/filters/BreakdownFilter/BreakdownTagMenu.tsx:137]: Indeed, the UX here is poor. I'll improve it in the next PR as it seems I made the breakdown pills huge now 
image

**Pull request merged**